examples/nlp/intent_slot_classification/data/import_datasets.py

```diff
@@ -0,0 +1,279 @@
+            processed_index = 0
+            for tag_start, tag_end, tag_str in slot_tags_list:
+                if tag_start > processed_index:
+                    words_list = sentence[processed_index:tag_start].strip().split()
+                    slots.extend([str(slots_list_all['O'])] * len(words_list))
+                words_list = sentence[tag_start:tag_end].strip().split()
+                slots.append(str(slots_list_all[f'B-{tag_str}']))
+                slots.extend([str(slots_list_all[f'I-{tag_str}'])] * (len(words_list) - 1))
+                processed_index = tag_end
+
+            if processed_index < len(sentence):
+                words_list = sentence[processed_index:].strip().split()
+                slots.extend([str(slots_list_all['O'])] * len(words_list))
+
+            slots = slots[1:-1]
+            slot = ' '.join(slots)
+            outfiles[mode + '_slots'].write(slot + '\n')
+
+        outfiles[mode + '_slots'].close()
+        outfiles[mode].close()
+
+    outfiles['dict_slots'].close()
+    outfiles['dict_intents'].close()
+
+    return outfold
+
+
+if __name__ == "__main__":
+    # Parse the command-line arguments.
+    parser = argparse.ArgumentParser(description="Process and convert datasets into NeMo\'s format.")
+    parser.add_argument(
+        "--dataset_name",
+        required=True,
+        type=str,
+        choices=['atis', 'snips', 'jarvis', 'dialogflow', 'mturk-processed', 'assistant'],
+    )
+    parser.add_argument(
+        "--source_data_dir", required=True, type=str, help='path to the folder containing the dataset files'
+    )
+    parser.add_argument("--target_data_dir", required=True, type=str, help='path to save the processed dataset')
+    parser.add_argument("--do_lower_case", action='store_true')
+    parser.add_argument(
+        "--ignore_prev_intent",
+        action='store_true',
+        help='ignores previous intent while importing datasets in jarvis\'s format',
+    )
+    parser.add_argument(
+        "--use_full_dataset",
+        action='store_true',
+        help='using full 25K dataset versus smaller 10K version for assistant\'s dataset',
+    )
+
+    args = parser.parse_args()
+
+    dataset_name = args.dataset_name
+    source_dir = args.source_data_dir
+    target_dir = args.target_data_dir
+
+    if not exists(source_dir):
+        raise FileNotFoundError(f"{source_dir} does not exist.")
+
+    if dataset_name == 'atis':
+        process_atis(infold=source_dir, outfold=target_dir, do_lower_case=args.do_lower_case)
+    elif dataset_name == 'snips':
+        process_snips(infold=source_dir, outfold=target_dir, do_lower_case=args.do_lower_case)
+    elif dataset_name == 'jarvis':
+        process_jarvis_datasets(
+            infold=source_dir,
+            outfold=target_dir,
+            modes=["train", "test", "dev"],
+            do_lower_case=args.do_lower_case,
+            ignore_prev_intent=args.ignore_prev_intent,
+        )
+    elif dataset_name == 'dialogflow':
```

Comment:
You may drop dialogflow and mturk here for now. But please check with Purnendu to see what these are. If we need them in nemo 1.0. We can add them.

Reply:
I already removed it.